src/spatialdata/_core/spatialdata.py

```diff
@@ -210,6 +210,84 @@ def from_elements_dict(elements_dict: dict[str, Union[SpatialElement, AnnData]])
     def query(self) -> QueryManager:
         return self._query
 
+    def aggregate(
+        self,
+        values: DaskDataFrame | GeoDataFrame | SpatialImage | MultiscaleSpatialImage,
+        by: str,
```

Comment:
I would either make values be a string, or make by be a `GeoDataFrame | Image2DModel | Image3DModel | Labels2DModel | Labels3DModel`. I prefer the second. The string can be then derived by calling `sdata._locate_spatial_element()`, as it is done for instance in the function `transform_to_coordinate_system()`.

Reply:
Ok I think I got it.